src/proc.cpp

```diff
@@ -244,10 +244,10 @@ void job_t::set_flag(job_flag_t flag, bool set) {
 bool job_t::get_flag(job_flag_t flag) const { return !!(this->flags & flag); }
 
 int job_signal(job_t *j, int signal) {
-    pid_t my_pid = getpid();
+    pid_t my_pgid = getpgrp();
     int res = 0;
 
-    if (j->pgid != my_pid) {
+    if (j->pgid != my_pgid) {
```

Comment:
Actually, it doesn't try hard enough! Again, it's possible for fish to not be the process group leader (e.g. it is a process in a pipeline), and it is technically possible for the process to be inadvertently signalled. It seems unlikely, but worth fixing.